repos/system_upgrade/el7toel8/actors/checkkdegnome/libraries/library.py

```diff
@@ -0,0 +1,158 @@
+        return last_modified >= int(time.time() - ONE_MONTH)
+    return False
+
+
+def is_installed(app):
+    """
+    Wrapper for "rpm -q <app>" command, returns True if application is found,
+    False in other cases. Output of rpm command is not supressed.
+    """
+    return True if subprocess.call(["rpm", "-q", app]) == 0 else False
+
+
+def check_kde_gnome():
+    apps_in_use = 0
+    api.current_logger().info("  Detecting desktop environments  ")
+    api.current_logger().info("==================================")
+
+    # Detect installed desktops by their startup files
+    kde_desktop_installed = is_executable("/usr/bin/startkde")
+    gnome_desktop_installed = is_executable("/usr/bin/gnome-session")
+    api.current_logger().info("* KDE installed: {0}".format(kde_desktop_installed))
+    api.current_logger().info("* Gnome installed: {0}".format(gnome_desktop_installed))
+    api.current_logger().info("----------------------------------")
+
+    # No desktop installed, we don't even care about apps as they are most likely not used or even installed
+    if not kde_desktop_installed and not gnome_desktop_installed:
+        api.current_logger().info("No desktop installed. Continuing with the upgrade.")
+        return
+
+    if kde_desktop_installed:
+        api.current_logger().info("KDE desktop is installed. Checking what we can do about it.")
+        if not gnome_desktop_installed:
+            api.current_logger().error("Cannot perform the upgrade because there is\
+             no other desktop than KDE installed.")
+            # We cannot continue with the upgrade process
+            reporting.create_report([
+                reporting.Title("Cannot upgrade because there is no other desktop than KDE installed."),
+                reporting.Summary("With only KDE installed, there would be no other desktop after upgrade."),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Tags([
+                    reporting.Tags.UPGRADE_PROCESS
+                ]),
+                reporting.Flags([
+                    reporting.Flags.INHIBITOR
+                ]),
+                reporting.Remediation(
+                    hint="Install GNOME desktop to be able to upgrade.")
+                ])
+            return
+
+        # Assume GNOME is installed in this state
+
+        user = os.environ.get("USER")
+        default_xsession = get_xsession("/var/lib/AccountsService/users/{0}".format(user))
+        if not default_xsession:
+            api.current_logger().warn("Unable to detect default session.")
+        else:
+            if "plasma" in default_xsession:  # using in because there can be some white spaces
+                api.current_logger().info("KDE used as default session.")
+                api.current_logger().info("Upgrade can be performed, but KDE desktop will\
+                 be removed in favor of GNOME")
```

Comment:
Again, instead of wrapping like that:
```python
api.current_logger().info("Upgrade can be performed, but KDE desktop will"
                                        "be removed in favor of GNOME")
```

Reply:
Changed everywhere.